src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -169,224 +176,668 @@ private sealed class ColInfoEx
+
+                var test = TestType(_srcTypes[iinfo]);
+                return GetTopic(input, iinfo);
+            }
+
+            private ValueGetter<VBuffer<Float>> GetTopic(IRow input, int iinfo)
+            {
+                var getSrc = RowCursorUtils.GetVecGetterAs<Double>(NumberType.R8, input, _srcCols[iinfo]);
+                var src = default(VBuffer<Double>);
+                var lda = _parent._ldas[iinfo];
+                int numBurninIter = lda.InfoEx.NumBurninIter;
+                bool reset = lda.InfoEx.ResetRandomGenerator;
+                return
+                    (ref VBuffer<Float> dst) =>
+                    {
+                        // REVIEW: This will work, but there are opportunities for caching
+                        // based on input.Counter that are probably worthwhile given how long inference takes.
+                        getSrc(ref src);
+                        lda.Output(in src, ref dst, numBurninIter, reset);
+                    };
+            }
+        }
+
+        public const string LoaderSignature = "LdaTransform";
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "LIGHTLDA",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(LdaTransform).Assembly.FullName);
+        }
+
+        private readonly TransformInfo[] _exes;
+        private readonly LdaState[] _ldas;
+        private readonly ColumnType[] _types;
+
+        private const string RegistrationName = "LightLda";
+        private const string WordTopicModelFilename = "word_topic_summary.txt";
+        internal const string Summary = "The LDA transform implements LightLDA, a state-of-the-art implementation of Latent Dirichlet Allocation.";
+        internal const string UserName = "Latent Dirichlet Allocation Transform";
+        internal const string ShortName = "LightLda";
+
+        private static (string input, string output)[] GetColumnPairs(ColumnInfo[] columns)
+        {
+            Contracts.CheckValue(columns, nameof(columns));
+            return columns.Select(x => (x.Input, x.Output)).ToArray();
+        }
+
+        internal LdaTransform(IHostEnvironment env, IDataView input, ColumnInfo[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(LdaTransform)), GetColumnPairs(columns))
+        {
+            _exes = new TransformInfo[columns.Length];
+            _types = new ColumnType[columns.Length];
+            _ldas = new LdaState[columns.Length];
+
+            for (int i = 0; i < columns.Length; i++)
+            {
+                var ex = new TransformInfo(Host, columns[i]);
+                _exes[i] = ex;
+                _types[i] = new VectorType(NumberType.Float, ex.NumTopic);
+            }
```

Comment:
You probably want to test the type before starting the training process.

Reply:
fixed

---
In reply to: [231652541](https://github.com/dotnet/machinelearning/pull/1410#discussion_r231652541) [](ancestors = 231652541)